src/magic.cpp

```diff
@@ -683,34 +744,52 @@ bool spell::bp_is_affected( body_part bp ) const
     return type->affected_bps[bp];
 }
 
-void spell::create_field( const tripoint &at ) const
+bool spell::create_field( const tripoint &at ) const
 {
     if( !type->field ) {
-        return;
+        return false;
     }
     const int intensity = field_intensity() + rng( -type->field_intensity_variance * field_intensity(),
                           type->field_intensity_variance * field_intensity() );
     if( intensity <= 0 ) {
-        return;
+        return false;
     }
-    if( one_in( type->field_chance ) ) {
+    if( one_in( type->field_chance ) == !type->field_chance_inv ) {
         field_entry *field = g->m.get_field( at, *type->field );
         if( field ) {
             field->set_field_intensity( field->get_field_intensity() + intensity );
         } else {
             g->m.add_field( at, *type->field, intensity );
         }
+        return true;
     }
+    return false;
 }
 
-void spell::make_sound( const tripoint &target ) const
+void spell::make_sound( const Creature &caster, const tripoint &target ) const
 {
     if( !has_flag( spell_flag::SILENT ) ) {
-        int loudness = abs( damage() ) / 3;
+        int loudness = abs( damage() ) / 3 ;
         if( has_flag( spell_flag::LOUD ) ) {
             loudness += 1 + damage() / 3;
         }
-        sounds::sound( target, loudness, sounds::sound_t::combat, _( "an explosion" ), false );
+        if( type->s_vol != 0) {
+            loudness += type->s_vol;
+        }
+
+        std::string msg = type->s_msg.empty() ? "an explosion!" : type->s_msg;
```

Comment:
This is missing the translation (`s_msg` and the string literal).

The whole string will be strange: "an explosion!Your Name magic wand" (with "Your Name" coming from `caster.disp_name`, "magic wand" from `obj_name` and  "an explosion!" being hardcoded.

There are at the very least some "connection" strings missing, or some spaces.

Reply:
The a ? b : c operator (Ternary conditional) should take b if a is true and c if a is false.

Reply:
True. It'll be reimplemented, but idK how the messaging will be handeld yet.